src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -260,15 +260,19 @@ class PopoverReportActionContextMenu extends React.Component {
 
     confirmDeleteAndHideModal() {
         this.callbackWhenDeleteModalHide = () => this.onComfirmDeleteModal = this.runAndResetCallback(this.onComfirmDeleteModal);
-        Report.deleteReportComment(this.state.reportID, this.state.reportAction);
+
+        if (this.state.reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU) {
+            // @TODO: Implement new DeleteMoneyRequest API command from issue https://github.com/Expensify/Expensify/issues/270502
+        } else {
+            Report.deleteReportComment(this.state.reportID, this.state.reportAction);
+        }
         this.setState({isDeleteCommentConfirmModalVisible: false});
     }
 
     hideDeleteModal() {
         this.callbackWhenDeleteModalHide = () => this.onCancelDeleteModal = this.runAndResetCallback(this.onCancelDeleteModal);
         this.setState({
             reportID: '0',
-            reportAction: {},
```

Comment:
I removed this to avoid a bug where the modal title and text updated before the dismiss animation finished. This doesn't seem to introduce any regressions

Reply:
This PR caused a [regression](https://github.com/Expensify/App/issues/18165) by removing the reportAction reset. For More details refer to this [comment](https://github.com/Expensify/App/issues/18165#issuecomment-1531179702)